cluster/grpc_rpc_client.go

```diff
@@ -241,3 +249,33 @@ func (gs *GRPCClient) BeforeShutdown() {}
 func (gs *GRPCClient) Shutdown() error {
 	return nil
 }
+
+func (gs *GRPCClient) getServerHost(sv *Server) (host, portKey string) {
+	var (
+		serverRegion, hasRegion   = sv.Metadata[constants.RegionKey]
+		externalHost, hasExternal = sv.Metadata[constants.GRPCExternalHostKey]
+		internalHost, _           = sv.Metadata[constants.GRPCHostKey]
+	)
+
+	hasRegion = hasRegion && serverRegion != ""
+	hasExternal = hasExternal && externalHost != ""
```

Comment:
What happens if the server doesn't specify external port but specifies external host?

Reply:
It'll fail here https://github.com/topfreegames/pitaya/blob/9bede6d29051472d79eade0f4a98076b12374fcc/cluster/grpc_rpc_client.go#L216

Reply:
It'll fail here: https://github.com/topfreegames/pitaya/blob/9bede6d29051472d79eade0f4a98076b12374fcc/cluster/grpc_rpc_client.go#L216